workers/unity/Assets/Playground/Scripts/Worker/WorkerConnectorBase.cs

```diff
@@ -0,0 +1,227 @@
+                        await Worker.CreateWorkerAsync(config, SelectDeploymentName, logger, origin)
+                            .ConfigureAwait(false);
+                    var worker =
+                        await ConnectWithRetries(connectionDelegate, MaxConnectionAttempts, logger, workerType);
+                    InitializeWorker(worker);
+                }
+                else
+                {
+                    var config = GetReceptionistConfig(workerType);
+                    ConnectionDelegate connectionDelegate = async () =>
+                        await Worker.CreateWorkerAsync(config, logger, origin).ConfigureAwait(false);
+                    var worker =
+                        await ConnectWithRetries(connectionDelegate, MaxConnectionAttempts, logger, workerType);
+                    InitializeWorker(worker);
+                }
+            }
+            catch (Exception e)
+            {
+                logger.HandleLog(LogType.Error, new LogEvent("Failed to create worker")
+                    .WithException(e)
+                    .WithField("WorkerType", workerType)
+                    .WithField("Message", e.Message));
+                connectionAttemptFinishedTask.SetException(e);
+                Dispose();
+                return;
+            }
+
+            connectionAttemptFinishedTask.SetResult(true);
+        }
+
+        protected virtual void AddWorkerSystems()
+        {
+        }
+
+        protected virtual string SelectDeploymentName(DeploymentList deployments)
+        {
+            return deployments.Deployments[0].DeploymentName;
+        }
+
+        private static async Task<Worker> ConnectWithRetries(ConnectionDelegate connectionDelegate, int attempts,
+            ILogDispatcher logger, string workerType)
+        {
+            while (attempts > 0)
+            {
+                try
+                {
+                    var worker = await connectionDelegate();
+                    return worker;
+                }
+                catch (ConnectionFailedException e)
+                {
+                    logger.HandleLog(LogType.Error,
+                        new LogEvent($"Failed attempt to create worker")
+                            .WithField("WorkerType", workerType)
+                            .WithField("Message", e.Message));
+                    attempts--;
+                }
+            }
+
+            throw new ConnectionFailedException(
+                $"Exceeded maximum connection attempts ({attempts})",
+                ConnectionErrorReason.ExceededMaximumRetries);
+        }
+
+        private bool ShouldUseLocator()
+        {
+            if (Application.isEditor)
+            {
+                return false;
+            }
+
+            var commandLineArguments = Environment.GetCommandLineArgs();
+            var commandLineArgs = CommandLineUtility.ParseCommandLineArgs(commandLineArguments);
```

Comment:
the other option would be to factor these two lines out as they are used in three places anyway and make that method `virtual`

Reply:
I'm not sure I follow. We should remove most of the command line stuff and replace it with something leaner, but I'm not sure what would need to be virtual here.

This was mostly copying the existing code. We can change how this is done here, but I consider that a drive by.